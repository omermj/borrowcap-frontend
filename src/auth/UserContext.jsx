import React from "react";

/** Context to provide currentUser thoughout the app */

const UserContext = React.createContext();

export default UserContext;
