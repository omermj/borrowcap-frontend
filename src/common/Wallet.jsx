import { Col, Container, Row } from "react-bootstrap";
import { Formik } from "formik";
import { Form, Button } from "react-bootstrap";
import BorrowcapApi from "../api/api";
import { formatCurrency } from "../helpers/format";
import { useState, useContext } from "react";
import UserContext from "../auth/UserContext";

/** Component for Wallet for Borrowers and Investors */

const Wallet = () => {
  const { currentUser } = useContext(UserContext);
  const [accountBalance, setAccountBalance] = useState(
    currentUser.accountBalance
  );

  return (
    <div className="form-wrapper mx-4">
      <Container>
        <Row>
          <Col>
            <h3 className="mb-4 text-center">
              Account Balance: {formatCurrency(accountBalance)}
            </h3>
          </Col>
        </Row>
        <Row>
          <Col className="form-inner">
            <h6 className="mb-3 text-center">Deposit Funds</h6>
            <Formik
              initialValues={{
                amount: "",
              }}
              onSubmit={async (values, { setSubmitting, setErrors }) => {
                values.id = currentUser.id;
                const res = await BorrowcapApi.depositFunds(
                  currentUser.id,
                  values
                );
                if (!!res) {
                  setAccountBalance(res.accountBalance);
                  values.amount = "";
                }
              }}
            >
              {({
                values,
                errors,
                handleChange,
                handleSubmit,
                isSubmitting,
              }) => (
                <Form onSubmit={handleSubmit}>
                  <Form.Group className="mb-2" controlId="depositFormAmount">
                    <Form.Control
                      type="number"
                      name="amount"
                      min={0}
                      value={values.amount}
                      onChange={handleChange}
                      placeholder="Enter deposit amount"
                      required
                    />
                  </Form.Group>
                  <div className="text-center">
                    <Button
                      variant="primary"
                      type="submit"
                      disabled={isSubmitting}
                      className="mt-3 px-4 btn-dark"
                    >
                      Deposit
                    </Button>
                  </div>
                </Form>
              )}
            </Formik>
          </Col>
          <Col className="form-inner mx-4">
            <h6 className="mb-3 text-center">Withdraw Funds</h6>
            <Formik
              initialValues={{
                amount: "",
              }}
              onSubmit={async (values, { setSubmitting, setErrors }) => {
                values.id = currentUser.id;
                const res = await BorrowcapApi.withdrawFunds(
                  currentUser.id,
                  values
                );
                if (!!res) {
                  setAccountBalance(res.accountBalance);
                  values.amount = "";
                }
              }}
            >
              {({
                values,
                errors,
                handleChange,
                handleSubmit,
                isSubmitting,
              }) => (
                <Form onSubmit={handleSubmit}>
                  <Form.Group className="mb-0" controlId="withdrawFormAmount">
                    <Form.Control
                      type="number"
                      name="amount"
                      min={0}
                      value={values.amount}
                      onChange={handleChange}
                      placeholder="Enter withdrawal amount"
                      required
                    />
                  </Form.Group>
                  <div className="text-center">
                    <Button
                      variant="primary"
                      type="submit"
                      disabled={isSubmitting}
                      className="mt-3 px-4 btn-dark"
                    >
                      Withdraw
                    </Button>
                  </div>
                </Form>
              )}
            </Formik>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default Wallet;
