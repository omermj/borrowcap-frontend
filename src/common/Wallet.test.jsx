import { render } from "@testing-library/react";
import Wallet from "./Wallet";
import { BrowserRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";

// smoke test
it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <UserContext.Provider value={{ currentUser: { username: "test" } }}>
        <Wallet />
      </UserContext.Provider>
    </BrowserRouter>
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <UserContext.Provider value={{ currentUser: { username: "test" } }}>
        <Wallet />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
