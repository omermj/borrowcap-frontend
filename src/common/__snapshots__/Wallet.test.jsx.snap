// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-wrapper mx-4"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="mb-4 text-center"
          >
            Account Balance: $20,000.00
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="form-inner col"
        >
          <h6
            class="mb-3 text-center"
          >
            Deposit Funds
          </h6>
          <form
            class=""
          >
            <div
              class="mb-2"
            >
              <input
                class="form-control"
                id="depositFormAmount"
                name="amount"
                placeholder="Enter deposit amount"
                type="number"
                value=""
              />
            </div>
            <div
              class="text-center"
            >
              <button
                class="mt-3 px-4 btn-dark btn btn-primary"
                type="submit"
              >
                Deposit
              </button>
            </div>
          </form>
        </div>
        <div
          class="form-inner mx-4 col"
        >
          <h6
            class="mb-3 text-center"
          >
            Withdraw Funds
          </h6>
          <form
            class=""
          >
            <div
              class="mb-2"
            >
              <input
                class="form-control"
                id="withdrawFormAmount"
                name="amount"
                placeholder="Enter withdrawal amount"
                type="number"
                value=""
              />
            </div>
            <div
              class="text-center"
            >
              <button
                class="mt-3 px-4 btn-dark btn btn-primary"
                type="submit"
              >
                Withdraw
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
